File: C#/0993. Cousins in Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsCousins(TreeNode root, int x, int y) {
        if(root==null){
            return false;
        }
        Queue<TreeNode> rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        while(rootQueue.Count()!=0){
            bool flag=false;
            int lenOfRootQueue=rootQueue.Count();
            for(int i=0;i<lenOfRootQueue;i++){
                TreeNode curRoot=rootQueue.Dequeue();
                if(curRoot.left!=null && curRoot.right!=null){
                    if(curRoot.left.val==x && curRoot.right.val==y){return false;}
                    if(curRoot.left.val==y && curRoot.right.val==x){return false;}
                }
                if(curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                    if((curRoot.left.val==x || curRoot.left.val==y) && flag==false){
                        flag=true;
                    }
                    else if((curRoot.left.val==x ||curRoot.left.val==y) && flag==true){
                        return true;
                    }
                }
                if(curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                    if((curRoot.right.val==x ||curRoot.right.val==y) && flag==false){
                        flag=true;
                    }
                    else if((curRoot.right.val==x ||curRoot.right.val==y) && flag==true){
                        return true;
                    }
                }
            }
        }
        return false;
    }
}

File: archives/contest/0993. Cousins in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isCousins(self, root: 'TreeNode', x: 'int', y: 'int') -> 'bool':
        if not root:
            return False
        rootlist=[root]
        while rootlist:
            level_val=set()
            len_rootlist=len(rootlist)
            for _ in range(len_rootlist):
                cur_root=rootlist.pop(0)
                level_val.add(cur_root.val)
                if cur_root.left and cur_root.right:
                    if (cur_root.left.val==x and cur_root.right.val==y) or (cur_root.left.val==y and cur_root.right.val==x):
                        return False
                if cur_root.left:
                    rootlist.append(cur_root.left)
                if cur_root.right:
                    rootlist.append(cur_root.right)
            
            if x in level_val and y in level_val:
                return True
        return False
